polish_example_json.py

```diff
@@ -2,148 +2,188 @@
+    :param schema: Dictionary parsed JSON schema
+    :return: Dictionary format polished example JSON
+    """
+    polished_example_json = example_json
+
+    if schema.get('type') == 'object':
+        # If given an object then recursively step into the properties
+        return polish_json_examples(example_json, schema['properties'])
+
+    for key, example_value in example_json.items():
+        if key.startswith('$') or not (key_schema := schema.get(key)):
+            # $ prefix indicates a JSON schema parameter or no reference to this key in the schema, ignore it
+            continue
+
+        if (schema_examples := key_schema.get('examples')) and (example_value not in schema_examples):
+            # If the schema provides examples and the existing example JSON does not adhere,
+            # then replace with a choice of schema example values
+            polished_example_json[key] = random.choice(schema_examples)
+
+        elif key_schema.get('type') == 'array':
+            # Recursively step through arrays.
+            polished_example_json[key] = [polish_json_examples(i, key_schema.get('items', {})) for i in example_value]
+
+        elif key_schema.get('type') == 'object':
+            # Recursively step into sub objects.
+            polished_example_json[key] = polish_json_examples(example_value, key_schema.get('properties', {}))
+
+    return polished_example_json
+
+
+def generate_fake_sample_data(survey_shape: Dict):
+    # Generate fake sample data for a specific survey shape.
+    # Note that the data may not conform to the sample spec, but the set of fields will at least be the correct shape
+    column_names = []
+    sensitive_column_names = []
+
+    for shape_config in survey_shape:
+        is_sensitive = shape_config.get('sensitive')
+        column_name = shape_config['columnName']
+        if is_sensitive:
+            sensitive_column_names.append(column_name)
+        else:
+            column_names.append(column_name)
+
+    fake_sample = {sample_key: f"dummy_{''.join(random.choices(string.ascii_lowercase + string.digits, k=10))}"
+                   for sample_key in column_names}
+
+    fake_sample_sensitive = {
+        sample_key: f"dummy_{''.join(random.choices(string.ascii_lowercase + string.digits, k=10))}"
+        for sample_key in sensitive_column_names}
+
+    fake_sample_sensitive_redacted = {
+        sample_key: "REDACTED" for sample_key in sensitive_column_names}
+    return fake_sample, fake_sample_sensitive, fake_sample_sensitive_redacted
+
+
+def to_camel_case(snake_str):
+    components = snake_str.split('-')
+    return reduce(lambda x, y: x + y.title(), components[1:], components[0])
+
 
-                with open(f'examples/{survey_type["type"]}/{event_item["event"]}.example.json', 'w+') as example_file:
-                    json.dump(event, example_file, indent=2)
+if __name__ == '__main__':
+    main()
```

Comment:
Highlighting these as the main, important changes to review as they are somewhat buried in the PR changes view.